### DISCUSSION TITLE
Using targets in an Rstdio Template

### DISCUSSION THREAD

**Discussion by Dpananos** [ropensci/targets#1258] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I'd like to use `targets` in an Rstudio template.  This would mean running `targets::use_targets` for a specific directory which us not the present working directory.

`renv` has the capability I'm looking for.  I can initiate an renv in a specified project through the `project` argument.  This will create an `renv` directory in the path passed through the `project1 argument.

Here is a simplified example of my existing template

```
# analysis_template.R
statistical_analysis_template <- function(path, ...) {

  dots <- list(...)
  # ensure path exists
  dir.create(path, recursive = TRUE, showWarnings = FALSE)

  # If choose to use renv, init renv in the project
  if(dots$use_renv) {
    renv::init(path)

    # If using targets, install targets to the path
    if(dots$use_targets){
      renv::install('targets', project=path, prompt = FALSE)
    }

  }

  # Finally, instantiate a targets
  if(dots$use_targets){
    ## Here is the problem.  I need to apply use_targets in the path
    # setwd does not work here.
    targets::use_targets(open = FALSE)
  }

}
```

and my dcf file

```
# statistical_analysis.dcf
Binding: statistical_analysis_template
Title: Statistical Analysis Template

Parameter: use_targets
Widget: CheckboxInput
Label: Use Targets
Default: On
Position: left

Parameter: use_renv
Widget: CheckboxInput
Label: Use Renv
Default: On
Position: left
```

**Comment by joelnitta** [comment, marked as answer]
What about `withr::with_dir()`?

For example, `withr::with_dir(path, targets::use_targets(open = FALSE))`

**Reply by Dpananos** [reply]
That's a good library to know about, thanks
